audio-worklet/src/subtractive-synth-demo.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+class SubtractiveSynthDemo {
+  constructor(context) {
+    this.maxCutoff_ = 5000;
+    this.masterGain_ = new GainNode(context, {gain: 0.5});
+
+    this.synthesizer_ = new Synthesizer(context, {
+      maxCutoff: this.maxCutoff_,
+    });
+
+    this.synthesizer_.output.connect(this.masterGain_)
+        .connect(context.destination);
+
+    // Initialize a QWERTY keyboard defined in qwerty-hancock.min.js
+    // The key A on a computer keyboard maps onto the note C, and the letters
+    // to the right of A (SDFGHJ) map onto the notes above C in a C major scale
+    // (DEFGAB). The letters WETYU map onto the black keys of a piano.
+    this.keyboard_ = new QwertyHancock(
+        {id: 'keyboard', width: 600, height: 150, octaves: 2});
+
+    this.keyboard_.keyDown = this.keyDown.bind(this);
+    this.keyboard_.keyUp = this.keyUp.bind(this);
+  }
+
+  /**
+   * Initialize GUI components
+   * @param {String} containerId the id of the HTML container
+   */
+  initializeGUI(containerId) {
+    this.lowPassCutoffSlider_ = new ParamController(
+        containerId, this.synthesizer_.setLowPass.bind(this.synthesizer_), {
+          type: 'range',
+          min: 0,
+          max: this.maxCutoff_,
+          step: 1,
+          default: this.maxCutoff_,
+          name: 'Low Pass Cutoff'
+        });
```

Comment:
It has an effect if you scrub it as you are playing a note, but currently it does not change the default lowpass cutoff for a voice, so you can't hear an effect by changing its values unless the voice is already active. Thats a bug which I am fixing in the next release.